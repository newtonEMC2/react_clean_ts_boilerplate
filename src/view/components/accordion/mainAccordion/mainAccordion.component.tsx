import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';

const useStyles = makeStyles({
  table: {
    minWidth: 650,
  },
});

export const SimpleAccordion: React.FC<{data:any}> = ({ data = [] }) => {
  const classes = useStyles();

  return (
    <TableContainer component={Paper}>
      <Table className={classes.table} aria-label="simple table">
        <TableHead>
          <TableRow>
            <TableCell>User Id</TableCell>
            <TableCell>title</TableCell>
            <TableCell>completed</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {data.map((row:any) => (
            <TableRow key={row.id}>
              <TableCell>{row.userId}</TableCell>
              <TableCell>{row.title}</TableCell>
              <TableCell>{row.completed.toString()}</TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </TableContainer>
  );
};
