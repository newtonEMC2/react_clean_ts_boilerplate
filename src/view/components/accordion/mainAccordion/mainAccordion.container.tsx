import React, { FC, useEffect } from 'react';

import { useDispatch, useSelector } from 'react-redux';

import { fetchUsers } from '../../../../application/actions/users.action';

import { SimpleAccordion } from './mainAccordion.component';

export const MainAccordionContainer: FC = () => {
  const users = useSelector((state:any) => state.users.collection) ?? [];
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchUsers());
  }, []);
  return (
    <SimpleAccordion data={users} />
  );
};
