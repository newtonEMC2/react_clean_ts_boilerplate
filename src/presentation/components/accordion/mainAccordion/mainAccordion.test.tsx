import React from 'react';
import { render } from '@testing-library/react';

import { SimpleAccordion } from './mainAccordion.component';

describe('<MainAccordionComponent />', () => {
  it('matches snapshot when empty', () => {
    const component = render(<SimpleAccordion data={[]} />);
    expect(component.container).toMatchSnapshot();
  });
  it('matches snapshot when data values are passed on', () => {
    const data = [{
      userId: 1,
      id: 1,
      title: 'delectus aut autem',
      completed: false,
    }];
    const component = render(<SimpleAccordion data={data} />);
    expect(component.container).toMatchSnapshot();
  });
  it('renders without crashing', () => {
    render(<SimpleAccordion data={[]} />);
  });
});
