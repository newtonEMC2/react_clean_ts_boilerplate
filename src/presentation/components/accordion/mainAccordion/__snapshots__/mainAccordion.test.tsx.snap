// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainAccordionComponent /> matches snapshot when data values are passed on 1`] = `
<div>
  <div
    class="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <table
      aria-label="simple table"
      class="MuiTable-root makeStyles-table-2"
    >
      <thead
        class="MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            User Id
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            title
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            completed
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            1
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            delectus aut autem
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            false
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`<MainAccordionComponent /> matches snapshot when empty 1`] = `
<div>
  <div
    class="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <table
      aria-label="simple table"
      class="MuiTable-root makeStyles-table-1"
    >
      <thead
        class="MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            User Id
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            title
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            completed
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root"
      />
    </table>
  </div>
</div>
`;
